### ISSUE TITLE
From normalized pixels values to original image dimensions

### ISSUE THREAD

**Issue by burhr2** [ultralytics/yolov5#2813]
## ❔Question
Hi! thanks for the great repo, I have tried several functions from the general.py to reverse my normalized [cx,cy,w,h] labels from the txt file to the original image dimensions but it seems I am not getting the original pixel values. Below is one of the function that I have tried: 

**What I want to do:**
reverse the normalized pixels in my training txt files to original pixels dimensions of the images
```
def xywhn2xyxy(x, image, padw=0, padh=0):
    # Convert nx4 boxes from [x, y, w, h] normalized to [x1, y1, x2, y2] where xy1=top-left, xy2=bottom-right
    w, h = image.shape[:2]
    y =  np.copy(x)
    y[0] = w * (x[0] - x[2] / 2) + padw  # top left x
    y[1] = h * (x[1] - x[3] / 2) + padh  # top left y
    y[2] = w * (x[0] + x[2] / 2) + padw  # bottom right x
    y[3] = h * (x[1] + x[3] / 2) + padh  # bottom right y
    return y
```
## Additional context
I have removed the tensors notations as my input to this function is just [cx,cy,w,h] and expects [x1,y1,x2,y2] which i will use to crop the object in the original image

**Comment by glenn-jocher** [comment]
@haspberry you can use scale_coords() from within detect.py as @GiorgioSgl mentioned, or you can also use YOLOv5 PyTorch Hub models to return detections in xyxy native image space:

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')

# Images
dir = 'https://github.com/ultralytics/yolov5/raw/master/data/images/'
imgs = [dir + f for f in ('zidane.jpg', 'bus.jpg')]  # batch of images

# Inference
results = model(imgs)
results.print()  # or .show(), .save()

print(results.pandas().xyxy[0])
```

```python
Downloading https://github.com/ultralytics/yolov5/releases/download/v5.0/yolov5s.pt to yolov5s.pt...
100%|██████████| 14.1M/14.1M [00:00<00:00, 22.7MB/s]
Adding autoShape... 
YOLOv5 🚀 2021-4-15 torch 1.8.1 CPU
image 1/2: 720x1280 2 persons, 2 ties
image 2/2: 1080x810 4 persons, 1 bus
Speed: 710.2ms pre-process, 275.3ms inference, 1.5ms NMS per image at shape (2, 3, 640, 640)

         xmin        ymin         xmax        ymax  confidence  class    name
0  749.472046   43.656982  1147.645874  704.565063    0.874176      0  person
1  433.666870  433.769470   517.574951  714.643982    0.688576     27     tie
2  115.053345  195.761230  1095.530273  708.221436    0.625422      0  person
3  986.198303  304.159058  1028.016602  420.192871    0.286980     27     tie
```

See PyTorch Hub Tutorial for details:

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [ONNX and TorchScript Export](https://docs.ultralytics.com/yolov5/tutorials/model_export)
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
